### DISCUSSION TITLE
Query Lists using CAML

### DISCUSSION THREAD

**Discussion by henningeiben** [pnp/pnpcore#541] [Q&A] [answered]
Hi,

I want to query a list multiple times - so I'm creating a CAML-query and then executing 

```c#
await myList.LoadItemsByCamlQueryAsync(viewXml)
var foundItems = myList.Items.AsRequested();
```

This works fine the first time, but when I repeat this with a modified viewXml (especially when viewXml will not return a result), my `foundItems` is still containing the result from the previous query. This doesn't seem right - or am I doing something wrong here?

**Comment by jansenbe** [comment, marked as answer]
@henningeiben: The method you've used is a "Load" method and the loaded data is kept in our domain model (see https://pnp.github.io/pnpcore/using-the-sdk/basics-getdata.html for more background). You can simply clear the model (`myList.Items.Clear()`) before the next attempt to get the desired result.

**Reply by henningeiben** [reply]
That's what I was missing!
